src/main/java/com/github/onsdigital/dpjavadurationparse/ParseDuration.java

```diff
@@ -0,0 +1,75 @@
+package com.github.onsdigital.dpjavadurationparse;
+
+import java.time.Duration;
+import java.time.format.DateTimeParseException;
+import java.util.HashMap;
+import java.util.Map;
+
+public class ParseDuration {
+
+    /**
+     * Parse's Duration in either ISO8601 or format's similar to: 3h56s345ms.
+     *
+     * The "PT" prefix on each of the ISO8601's durations example below, where P is
+     * a "period" duration designation ("P") and a "time" designation ("T") per the ISO-8601Checks whether duration
+     * starts with string looks like an ISO8601 formatted string, and if not it adds PT.
+     *
+     * @param stringDuration Checking whether {@link String} starts with the appropriate ISO8601 characters.
+     * @return A {@link Duration}
+     * Compatible with one or a combination of the following units: h, m, s
+     *
+     * Example Valid Formats
+     *          PT3h45m56s345ms
+     *          -PT3h45m56s345ms
+     *          PT3h56s345ms
+     *          -PT3h56s
+     *          PT3h
+     *          -PT56s
+     *
+     *          3h45m56s345ms34us
+     *          -3h45m56s345ms34µs
+     *          3h56s345ms34µs
+     *          -3h56s34µs
+     *          3h34µs
+     *          -56s
+     *
+     * Example Invalid Formats, resulting in null bring returned.
+     *          3h56s345ms34µs34us
+     *          34µs3h56s345ms
+     *          3h56s345ms34µs*
+     */
+    /*
+
+     */
+    public static Duration parseThisDuration(String stringDuration) {
+        // Flag to identify durations in the past
+        boolean negative = false;
+        // Quality check the quality of the parameter
+        if(stringDuration==null
+                || (stringDuration.startsWith("PT") && stringDuration.length()<4)                  // e.g. "PR1"
+                || (!stringDuration.startsWith("PT") && stringDuration.length()<2)) return null;   // e.g. "1"
+        stringDuration = stringDuration.toUpperCase();
+        // Check if it starts with a negative.
+        if(stringDuration.startsWith("-")) {
+            negative = true;
+            //remove the negative sign atm to allow it to be added back on later
+            stringDuration = stringDuration.substring(1);
+        }
+        // Check if user hasn't included the 'PT', and if they havn't then add it.
+        if (!stringDuration.startsWith("PT")) {
+            stringDuration = "PT".concat(stringDuration);
+        }
+        // Add the negative on if it was negative
+        if(negative) {
+            stringDuration = "-".concat(stringDuration);
+        }
+        // Try parse the duration.
+        Duration parsedDuration=null;
+        try {
+            parsedDuration = Duration.parse(stringDuration);
+        } catch (DateTimeParseException ex) {
+            return null;
```

Comment:
Have we considered throwing an exception if the string format is wrong? Or maybe returning an `Optional` instead of allowing null values